Userland/Applications/Help/MainWidget.cpp

```diff
@@ -99,28 +99,34 @@ MainWidget::MainWidget()
     m_web_view = find_descendant_of_type_named<WebView::OutOfProcessWebView>("web_view");
     m_web_view->on_link_click = [this](auto& url, auto&, unsigned) {
         if (url.protocol() == "file") {
-            auto path = url.path();
-            if (!path.starts_with("/usr/share/man/"sv)) {
+            auto path = LexicalPath { url.path() };
+            if (!path.is_child_of(Manual::manual_base_path)) {
                 open_external(url);
                 return;
             }
-            auto browse_view_index = m_manual_model->index_from_path(path);
+            auto browse_view_index = m_manual_model->index_from_path(path.string());
             if (browse_view_index.has_value()) {
                 dbgln("Found path _{}_ in m_manual_model at index {}", path, browse_view_index.value());
                 m_browse_view->selection().set(browse_view_index.value());
                 return;
             }
-            m_history.push(path);
-            open_page(path);
+            m_history.push(path.string());
+            open_page(path.string());
         } else if (url.protocol() == "help") {
             if (url.host() == "man") {
                 if (url.paths().size() != 2) {
                     dbgln("Bad help page URL '{}'", url);
                     return;
                 }
                 auto const section = url.paths()[0];
+                auto maybe_section_number = section.to_uint();
+                if (!maybe_section_number.has_value()) {
+                    dbgln("Bad section number '{}'", section);
+                    return;
+                }
+                auto section_number = maybe_section_number.value();
                 auto const page = url.paths()[1];
-                open_url(URL::create_with_file_scheme(String::formatted("/usr/share/man/man{}/{}.md", section, page), url.fragment()));
+                open_url(URL::create_with_file_scheme(make_ref_counted<Manual::PageNode>(Manual::sections[section_number - 1], page)->path(), url.fragment()));
```

Comment:
Are we immediately discarding this `PageNode`? I wonder whether there's a better way of doing this.

Reply:
You can either have a POD class that's nice to use here, or you can have the complex section cross-linking that Help needs. If you want the second one, you need ref counted classes unless you want to make a mess of data ownership.

Reply:
Fair enough, though I wonder whether we can't just hold the page objects directly instead of constructing them from the section on-demand. I might be missing some context.

Reply:
I'm pretty sure we can't create them outside of the callback unless we store them in a class member or else they will be out of scope + destructed by the time the callback runs.